### PR TITLE
Use fluentd core retry mechanism

### PR DESCRIPTION
**Pull request by yann-soubeyrand**
Based on PR #518, this PR disables ES client retrying mechanism and suppress ping at ES client creation to further reduce load on ES cluster.

### CODE REVIEW

**Comment by cosmo0920** [comment]
Whoa! It's great work. :smile: 
Could you rebase off master? I don't want to contaminated PR merging.

**Comment by yann-soubeyrand** [comment]
Hum, don't know why the test test_request_forever fails only with ruby version 24-x64 :-/

File: lib/fluent/plugin/out_elasticsearch.rb [hunk 1]
```diff
@@ -351,7 +346,6 @@ def client
                                                                               reload_connections: local_reload_connections,
                                                                               reload_on_failure: @reload_on_failure,
                                                                               resurrect_after: @resurrect_after,
-                                                                              retry_on_failure: 5,
```
Comment (cosmo0920) [inline lib/fluent/plugin/out_elasticsearch.rb]: :+1:

File: lib/fluent/plugin/elasticsearch_index_template.rb [hunk 2]
```diff
@@ -31,7 +34,7 @@ def retry_install(max_retries)
     retries = 0
     begin
       yield
-    rescue Fluent::Plugin::ElasticsearchOutput::ConnectionFailure, Timeout::Error => e
+    rescue *client.transport.host_unreachable_exceptions, Timeout::Error => e
```
Comment (cosmo0920) [inline lib/fluent/plugin/elasticsearch_index_template.rb]: :+1:

File: lib/fluent/plugin/out_elasticsearch_dynamic.rb [hunk 3]
```diff
@@ -50,7 +50,6 @@ def client(host = nil)
                                                                               reload_connections: @reload_connections,
                                                                               reload_on_failure: @reload_on_failure,
                                                                               resurrect_after: @resurrect_after,
-                                                                              retry_on_failure: 5,
```
Comment (cosmo0920) [inline lib/fluent/plugin/out_elasticsearch_dynamic.rb]: :+1:

**Comment by cosmo0920** [comment]
> Hum, don't know why the test test_request_forever fails only with ruby version 24-x64 :-/

I decided to disable these unstable tests for retry_forever in CI.

**Comment by cosmo0920** [comment]
I'll release this change as v3.0.0.
Becuase @yann-soubeyrand 's change is completely rewriting retrying mechanism and drops fluent-plugin-forest support in #517, #518.
It's time to bump up major version.

**Pull request merged**